libvast/native-plugins/summarize.cpp

```diff
@@ -750,10 +771,40 @@ class plugin final : public virtual pipeline_operator_plugin {
 
   [[nodiscard]] caf::expected<std::unique_ptr<pipeline_operator>>
   make_pipeline_operator(const record& config) const override {
-    auto parsed_config = configuration::make(config);
-    if (!parsed_config)
-      return parsed_config.error();
-    return std::make_unique<summarize_operator>(std::move(*parsed_config));
+    std::vector<std::string> sections_to_reformat;
+    for (const auto& key : {"min", "max", "any", "all", "sum"}) {
+      if (config.contains(key))
+        sections_to_reformat.emplace_back(key);
+    }
+    // new format detected. Proceed as usual
+    if (sections_to_reformat.empty()) {
+      return make_summarize_operator(config);
+    }
+    if (config.contains("aggregate")) {
+      return caf::make_error(ec::invalid_configuration,
+                             fmt::format("Aggregate section found in legacy "
+                                         "config"));
+    }
+    // Reformat old config into new one.
+    auto new_config = config;
+    auto aggregate = record{};
+    for (const auto& section : sections_to_reformat) {
+      VAST_WARN("Legacy format detected in summarize section: {}. Please "
+                "upgrade to the newest format",
+                section);
```

Comment:
```suggestion
VAST_WARN("the configuration key '{}' for the summarize operator is deprecated and will be removed in the next major release; please migrate to using explicit output field names", section);
```

Modulo formatting.

Reply:
This was moved because there was a force-push in the middle of my review, but the comment still applies.

Reply:
Done